### DISCUSSION TITLE
How are the lyrics synchronized with the music?

### DISCUSSION THREAD

**Discussion by Terkyz** [maxrave-dev/SimpMusic#163] [General]
I'm not an experienced programmer and I barely know a bit of Python (and I mention it in case it's necessary to explain technical details), but something that really fascinates me about SimpMusic is how well the lyrics of the song are synchronized with the song itself, how is it possible to do it with such precision?

I mean, it occurs to me that one of the backends that SimpMusic uses to represent the lyrics has some kind of marker where it says "This sentence is heard at minute 01:10 to 01:13" and once that marker ends the next one follows, but I have no idea, I really don't know and it seems to me something curious.

**Comment by maxrave-dev** [comment]
> I'm not an experienced programmer and I barely know a bit of Python (and I mention it in case it's necessary to explain technical details), but something that really fascinates me about SimpMusic is how well the lyrics of the song are synchronized with the song itself, how is it possible to do it with such precision?
> 
> I mean, it occurs to me that one of the backends that SimpMusic uses to represent the lyrics has some kind of marker where it says "This sentence is heard at minute 01:10 to 01:13" and once that marker ends the next one follows, but I have no idea, I really don't know and it seems to me something curious.

Sorry for late reply, check Musixmatch API for more information https://developer.musixmatch.com/documentation/api-reference/track-subtitle-get
